tests/test_workers.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+import signal
+import subprocess
+import sys
+import tempfile
+import time
+from typing import TYPE_CHECKING
+
+import httpx
+import pytest
+
+if TYPE_CHECKING:
+    from ssl import SSLContext
+    from typing import IO, Generator
+
+    from asgiref.typing import (
+        ASGIReceiveCallable,
+        ASGISendCallable,
+        LifespanStartupFailedEvent,
+        Scope,
+    )
+
+pytestmark = pytest.mark.skipif(sys.platform == "win32", reason="requires unix")
+gunicorn_arbiter = pytest.importorskip("gunicorn.arbiter", reason="requires gunicorn")
+uvicorn_workers = pytest.importorskip("uvicorn.workers", reason="requires gunicorn")
+
+
+class Process(subprocess.Popen):
+    client: httpx.Client
+    output: IO[bytes]
+
+    def read_output(self) -> str:
+        self.output.seek(0)
+        return self.output.read().decode()
```

Comment:
This is a little subclass of `subprocess.Popen` with some attributes added that are helpful for testing. In the future, we could trick this out more by subclassing [`asyncio.SubprocessProtocol`](https://docs.python.org/3/library/asyncio-protocol.html#asyncio.SubprocessProtocol) or [`anyio.abc.Process`](https://anyio.readthedocs.io/en/stable/api.html#anyio.abc.Process).

Reply:
This is fine.